src/routes/index.js

```diff
@@ -35,6 +35,10 @@ type RoutesProps = {
 const Routes = ({ defaultSafe, location }: RoutesProps) => {
   const [isInitialLoad, setInitialLoad] = useState<boolean>(true)
 
+  if (location.search.replace('?', '').split('&').includes('secret3BoxIntegration=true')) {
```

Comment:
hmm it would fire on every render, no? shouldn't this be a useEffect with empty dep array? or you can reuse isInitialLoad state var from above

Reply:
and also this would be true for non root route